classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -377,12 +407,23 @@ private function _migrateSubmissionFiles() {
 				'path' => $path,
 				'mimetype' => $row->file_type,
 			], 'file_id');
+			$fileId = $fileIdMap[$row->file_id] ?? $row->file_id;
+			$updateData = ['new_file_id' => $newFileId];
+			// If the file_id was forked into a new one, we've got to update the references at the review_round_files and the submission_file itself
+			if ($fileId !== $row->file_id) {
+				$updateData['file_id'] = $fileId;
+				$updateData['source_file_id'] = $fileIdMap['sourceFileId'];
```

Comment:
Not sure if these `submission_files` must be connected, but as they were connected before, I've decided to proceed with this.